cmd/sonobuoy/app/operations/cp.go

```diff
@@ -17,10 +17,148 @@ limitations under the License.
 package operations
 
 import (
+	"archive/tar"
 	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	podName       = "sonobuoy"
+	podNamespace  = "heptio-sonobuoy"
+	containerName = "kube-sonobuoy"
+	results       = "/tmp/sonobuoy"
 )
 
-func CopyResults(path string) error {
-	// pretty straight forwards
-	return errors.New("not implemented")
+// CopyResults copies results from a sonobuoy run into a Reader in tar format.
+func CopyResults(config *rest.Config, cmderr io.Writer, errc chan error) io.Reader {
```

Comment:
We'll have to pass in namespace and the other consts can be in config

Reply:
Why the error channel? Are we going to be reporting many errors?

Reply:
The error channel is required for capturing errors in the streaming go routine. CopyResults sets up the request, launches a go routine which streams the data and immediately returns the reader where the data being streamed will be accessible. If the goroutine gets an error our command should overall fail (exit 1).